Modules/IO/NIFTI/test/itkNiftiLargeImageRegionReadTest.cxx

```diff
@@ -0,0 +1,72 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         https://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#include "itkImageFileReader.h"
+#include "itkImageFileWriter.h"
+#include "itkTestingMacros.h"
+
+
+int
+itkNiftiLargeImageRegionReadTest(int argc, char * argv[])
+{
+
+  if (argc != 2)
+  {
+    std::cerr << "Missing arguments" << std::endl;
+    std::cerr << "Usage: " << itkNameOfTestExecutableMacro(argv) << " outputFileName" << std::endl;
+    return EXIT_FAILURE;
+  }
+
+
+  constexpr int Dimension = 3;
+  using PixelType = unsigned short;
+
+  using ImageType = itk::Image<PixelType, Dimension>;
+
+  // Create a large image
+  ImageType::SizeType   size = { { 1034, 1034, 1020 } };
+  ImageType::RegionType region;
+  region.SetSize(size);
+
+  auto image = ImageType::New();
+  image->SetRegions(region);
+  image->Allocate(true);
+
+  const std::string fname{ argv[1] };
+  itk::WriteImage(image, fname);
+
+
+  using ReaderType = itk::ImageFileReader<ImageType>;
+  auto reader = ReaderType::New();
+
+  reader->SetFileName(fname);
+
+  // Ensure that a sufficiently large subregion can be read
+  region.SetSize({ { 1029, 1029, 1017 } });
```

Comment:
Sorry Jon, would it still be possible to add a comment that explains why specifically this size `{ 1029, 1029, 1017 }` was chosen? I believe these numbers are not entirely arbitrary. `{ 1000, 1000, 1000 }` would also be large, right? Then why not `{ 1000, 1000, 1000 }`?!?  I think that's because the reading needs to be tested a buffer of more than `INT_MAX` bytes, right? 

If so, I think the comment could be something like:

    // Ensure that a sufficiently large subregion can be read, which would need the allocation a pixel buffer larger than
    // INT_MAX bytes. Previous versions of the nifti_clib (before 19 September 2022) did not yet support reading such
    // large subregions.    
 

What do you think? I hope I'm not asking too much  😃 !

Reply:
Done. Slightly reworded.